### PR TITLE
feat(ScrollTo): use fulls URLs, allow replace and render anchor links

### PR DESCRIPTION
**Pull request by nantunes**

### CODE REVIEW

**Review by MEsteves22 — APPROVED** [review]
LGTM 🙏

I just had the following questions/doubts while reviewing:
- The scroll item default role changed from `button` to `link`. I think this can be considered a breaking change. However, this change/fix makes sense to me since `href` defaults to `true`. Was this intentional? 
- `relativeLinks` defaults to `false` but I believe that to maintain the previous behaviour it should default to `true`. Can this change lead to any breaking change to the users that I'm not seeing?

**Review by zettca — COMMENTED** [review]
Looks good overall 👍 

The `navigationMode` feels like it should be the router's responsibility, but this is a good compromise.

A few notes regarding API & docs:

File: packages/core/src/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.tsx [hunk 1]
```diff
@@ -54,16 +67,19 @@ export const HvHorizontalScrollListItem = (
   const buttonId = setId(id, "button");
   const Tooltip = tooltipWrapper;
 
+  const Component = link != null ? "a" : "div";
```
[the remark this answers is not in the document]
Comment (nantunes) [reply packages/core/src/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.tsx:70]: Yes, I agree but kept the changes to the minimum in the PR.

File: packages/core/src/ScrollTo/Horizontal/ScrollToHorizontal.tsx [hunk 2]
```diff
@@ -104,19 +139,20 @@ export const HvScrollToHorizontal = (props: HvScrollToHorizontalProps) => {
 
   const elementId = useUniqueId(id, "hvHorizontalScrollto");
 
-  const [selectedIndex, setScrollTo] = useScrollTo(
+  const [selectedIndex, setScrollTo, elements] = useScrollTo(
     defaultSelectedIndex,
     scrollElementId,
-    href,
+    navigationMode,
+    relativeLinks,
```
[the remark this answers is not in the document]
Comment (nantunes) [reply packages/core/src/ScrollTo/Horizontal/ScrollToHorizontal.tsx:146]: I changed it under the assumption it is internal.

File: packages/core/src/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.tsx [hunk 3]
```diff
@@ -24,18 +25,29 @@ export interface HvVerticalScrollListItemProps extends HvBaseProps {
...
+  ) => void;
   /** A Jss Object used to override or extend the styles applied. */
   classes?: HvHorizontalScrollListItemClasses;
+
+  /**
+   * The link to be set in the href attribute of the anchor element.
+   *
+   * If this is not set, the element will be rendered as a div with a button role.
+   */
+  link?: string;
```
[the remark this answers is not in the document]
Comment (nantunes) [reply packages/core/src/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.tsx:43]: Actually it is `href` we are using as boolean, but it's ok. this is an internal component, no need to overthink it.

[the remark this answers is not in the document]
Comment (zettca) [reply packages/core/src/ScrollTo/Horizontal/HorizontalScrollListItem/HorizontalScrollListItem.tsx:43]: Yes, `href: boolean` is much worse - thanks for deprecating it 👍 😄

File: packages/core/src/ScrollTo/Horizontal/ScrollToHorizontal.test.tsx [hunk 4]
```diff
@@ -21,19 +21,19 @@ const Sample = () => (
 );
 
 describe("ScrollToHorizontal", () => {
-  it("renders the section buttons", () => {
+  it("renders the section links", () => {
     render(<Sample />);
 
-    const tab1 = screen.getByRole("button", { name: "Server status summary" });
+    const tab1 = screen.getByRole("link", { name: "Server status summary" });
```
[the remark this answers is not in the document]
Comment (zettca) [reply packages/core/src/ScrollTo/Horizontal/ScrollToHorizontal.test.tsx:27]: It's here: https://github.com/lumada-design/hv-uikit-react/blob/master/docs/overview/community/ReleaseVersioning.stories.mdx#L35-L39

> DOM or relevant accessibility tree changes

**Review by zettca — APPROVED** [review]
🚀

**Pull request merged**